web/src/components/dashboard/SidebarCardFundingNudge.test.tsx

```diff
@@ -63,27 +74,115 @@ describe("<SidebarCardFundingNudge />", () => {
     card = generateSidebarCardContent({ id: fundingNudge });
   });
 
+  const generateTaxFilingUserData = (params: {
+    publicFiling: boolean;
+    formedInNavigator?: boolean;
+    businessName?: string;
+    responsibleOwnerName?: string;
+    taxId?: string;
+    industryId?: string;
+  }): UserData => {
+    let formationData: FormationData = generateEmptyFormationData();
+
+    const legalStructureId = params.publicFiling
+      ? randomPublicFilingLegalType()
+      : randomLegalStructure({ requiresPublicFiling: false }).id;
+    if (params.publicFiling) {
+      formationData = generateFormationData(
+        {
+          completedFilingPayment: !!params.formedInNavigator,
+          getFilingResponse: generateGetFilingResponse({ success: params.formedInNavigator }),
+        },
+        legalStructureId as FormationLegalType
+      );
+    }
+
+    return generateUserDataForBusiness(
+      generateBusiness({
+        profileData: generateProfileData({
+          legalStructureId: legalStructureId,
+          operatingPhase: randomElementFromArray(
+            OperatingPhases.filter((obj) => {
+              return obj.displayTaxAccessButton;
+            })
+          ).id,
+          taxId: params.taxId ? `*${params.taxId.slice(1)}` : "",
+          encryptedTaxId: params.taxId ? `encrypted-${params.taxId}` : "",
+          businessName: params.businessName || "",
+          responsibleOwnerName: params.responsibleOwnerName || "",
+          industryId: params.industryId || "generic",
+        }),
+        formationData: formationData,
+        taxFilingData: generateTaxFilingData({ state: undefined }),
+      })
+    );
+  };
+
+  const userDataWithPrefilledFields = generateTaxFilingUserData({
+    publicFiling: true,
+    taxId: "123456789098",
+    businessName: "MrFakesHotDogBonanza",
+    industryId: randomIndustry().id,
+  });
+
+  const userDataWithGenericIndustry = generateTaxFilingUserData({
+    publicFiling: true,
+    taxId: "123456789123",
+    businessName: "MrFakesHotDog",
+    industryId: "generic",
+  });
+
   describe("when clicking funding button for non-generic industry", () => {
-    it("updates operating phase to UP_AND_RUNNING and marks tax registration task complete", () => {
+    it("updates operating phase to UP_AND_RUNNING and marks tax registration task complete", async () => {
```

Comment:
[pebble] We should add a test case for when the user progresses to the `UP_AND_RUNNING` phase and the tax registration task does not complete.

Ideally we'd cover cases where the user provides tax data and a negative response comes from the API and a case where they do not provide tax data.

Reply:
Added the jest tests to make sure that the `UP_AND_RUNNING` phase is set even when the tax registration tax does not complete. I did have to re-add the `mockAPI` functionality to mock the `taxFilingData` being set.